### PR TITLE
Adds halyard getting started codelab

### PR DESCRIPTION
**Pull request by skim1420**
@spinnaker/google FYI

### CODE REVIEW

File: guides/tutorials/codelabs/halyard-getting-started/index.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+### ~/.kube/config
+
+If you don't already have this, an easy way to set this up is to use gcloud:
+
+    gcloud container clusters get-credentials {YOUR_GKE_CLUSTER_NAME}
+
+### GCP service account
+
+You will need to specify the location of a json file containing credentials to your GCP project with the role "Storage / Storage Admin". Download the key from the Cloud Console and copy it to `~/.google/account.json`.
```
Comment (skim1420) [inline guides/tutorials/codelabs/halyard-getting-started/index.md]: @ewiseblatt This service account will be used for both front50/GCS as well as GCR. Not sure if there's a lesser privilege that will suffice for the needs.

File: guides/tutorials/codelabs/halyard-getting-started/index.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+## Part 1: Install halyard
+
+### Create halyard host VM
+
+```
+HALHOST=$USER-halyard-`date +%m%d`
+
+gcloud compute instances create $HALHOST \
+    --scopes storage-full,compute-rw \
+    --project={YOUR_GCP_PROJECT} \
```
Comment (lwander) [inline guides/tutorials/codelabs/halyard-getting-started/index.md]: I think it's OK to drop this, or to add a call like this

```
PROJECT=$(gcloud info --format='value(config.project)')
```

to make this guide more copy/paste-able

File: guides/tutorials/codelabs/halyard-getting-started/index.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+## Part 1: Install halyard
+
+### Create halyard host VM
+
+```
+HALHOST=$USER-halyard-`date +%m%d`
+
+gcloud compute instances create $HALHOST \
+    --scopes storage-full,compute-rw \
```
Comment (lwander) [inline guides/tutorials/codelabs/halyard-getting-started/index.md]: I don't think any scopes are needed on this VM if you're just running Halyard to deploy remotely.

File: guides/tutorials/codelabs/halyard-getting-started/index.md (hunk 1 above)
Comment (lwander) [reply guides/tutorials/codelabs/halyard-getting-started/index.md]: If the VM has the right scopes, you can download the keys straight to the VM with the commands shown here: https://spinnaker.github.io/setup/providers/docker-registry/#google-container-registry

File: guides/tutorials/codelabs/halyard-getting-started/index.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+. ~/.bashrc
+```
+
+## Part 2: Gather needed credentials
+
+### ~/.kube/config
+
+If you don't already have this, an easy way to set this up is to use gcloud:
+
```
Comment (lwander) [inline guides/tutorials/codelabs/halyard-getting-started/index.md]: Don't forget to add 
```
gcloud config set container/use_client_certificate true
```
first

File: guides/tutorials/codelabs/halyard-getting-started/index.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    gcloud container clusters get-credentials {YOUR_GKE_CLUSTER_NAME}
+
+### GCP service account
+
+You will need to specify the location of a json file containing credentials to your GCP project with the role "Storage / Storage Admin". Download the key from the Cloud Console and copy it to `~/.google/account.json`.
+
+
+## Part 3: Set Spinnaker configuration
+
+We will install Spinnaker v0.1.0
```
Comment (lwander) [inline guides/tutorials/codelabs/halyard-getting-started/index.md:171]: (Optimist Pram)

**Review by lwander — APPROVED** [review]
Thanks for setting this up - I'm wondering if instead we should move this under "quickstarts" or something in "setup", since that is probably where the user will look for something like this first.

File: guides/tutorials/codelabs/halyard-getting-started/index.md (hunk 1 above)
Comment (ewiseblatt) [reply guides/tutorials/codelabs/halyard-getting-started/index.md]: I'm not following this discussion or concerns.

Aside, where does ~/.google/account.json come from or get used?
I have seen "standard" places for credentials, but that isnt one I recall.
We dont use this, but the Google client libraries actually defines an environment variable GOOGLE_APPLICATION_CREDENTIALS that points to application default credentials.

It looks like you are putting this here so you can register it later. That's fine. The .google directory sounds like a standard and "account.json" sounds pretty generic, but it's purpose is quite specific.

BTW, the file should be made secure
chmod 400 ~/.google/account.json

**Comment by skim1420** [comment]
All comments addressed

**Pull request merged**